service/dap/server.go

```diff
@@ -993,10 +992,36 @@ func (s *Server) convertVariable(v *proc.Variable) (value string, variablesRefer
 	return
 }
 
-// onEvaluateRequest sends a not-yet-implemented error response.
+// onEvaluateRequest handles 'evalute' requests.
 // This is a mandatory request to support.
-func (s *Server) onEvaluateRequest(request *dap.EvaluateRequest) { // TODO V0
-	s.sendNotYetImplementedErrorResponse(request.Request)
+func (s *Server) onEvaluateRequest(request *dap.EvaluateRequest) {
+	showErrorToUser := request.Arguments.Context != "watch"
+	if s.debugger == nil {
+		s.sendErrorResponseWithOpts(request.Request, UnableToEvaluateExpression, "Unable to evaluate expression", "debugger is nil", showErrorToUser)
+		return
+	}
+	// Default to the topmost stack frame of the current goroutine in case
+	// no frame is specified (e.g. when stopped on entry or no call stack frame is expanded)
+	goid, frame := -1, 0
+	sf, ok := s.stackFrameHandles.get(request.Arguments.FrameId)
```

Comment:
Nit: unless you use `sf` outside it, it's more idiomatic to fold this into a single line `if`:

```
if sf, ok := ...; ok {
}
```